src/Storages/ObjectStorage/Azure/Configuration.cpp

```diff
@@ -279,6 +286,14 @@ void StorageAzureConfiguration::fromAST(ASTs & engine_args, ContextPtr context,
     }
 
     blobs_paths = {blob_path};
+    if (account_name && account_key)
```

Comment:
This is a minor, not necessary, feel free not to implement it. But what if you made `account_name` and `account_key` members of `StorageAzureConfiguration`? It is more clear & readable than `saved_params` imho.